src/drawer/component/internal/internal.component.ts

```diff
@@ -0,0 +1,203 @@
+
+@Component({
+  templateUrl: './internal.component.html',
+  styleUrls: ['./internal.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  standalone: true,
+  imports: [
+    NgIf,
+    NgClass,
+    NgStyle,
+    NgTemplateOutlet,
+    IconComponent,
+    CdkScrollable,
+    PortalModule,
+  ],
+  animations: [
+    trigger('showHide', [
+      state(
+        'show',
+        style({
+          opacity: 1,
+          transform: 'translateX(0)',
+        }),
+      ),
+      state(
+        'hide, void',
+        style({
+          opacity: 0,
+          transform: 'translateX(100%)',
+        }),
+      ),
+      transition('hide => show, void => show', [
+        animate(`${duration} ${TimingFunction.easeOut}`),
+      ]),
+      transition('show => hide, show => void', [
+        animate(`${duration} ${TimingFunction.easeInOut}`),
+      ]),
+    ]),
+  ],
+})
+export class DrawerInternalComponent<T = unknown, C = unknown> {
+  @ViewChild(CdkPortalOutlet, { static: false })
+  bodyPortalOutlet: CdkPortalOutlet;
+
+  @ViewChild('mask')
+  mask: ElementRef<HTMLElement>;
+
+  animationStep$ = new Subject<Step>();
+
+  options: DrawerOptions<T, C>;
+  showHide = 'hide';
+  maskVisible: boolean;
+
+  get drawerClasses(): Record<string, boolean> {
+    return {
+      'aui-drawer': true,
+      hasDivider: this.options.divider,
+      ...(this.options.drawerClass
+        ? { [this.options.drawerClass]: true }
+        : null),
+    };
+  }
+
+  get width() {
+    return handlePixel(
+      this.options.width || SIZE_MAPPER[this.options.size || DrawerSize.Medium],
+    );
+  }
+
+  isTemplateRef = isTemplateRef;
+
+  constructor(
+    private readonly cdr: ChangeDetectorRef,
+    private readonly injector: Injector,
+  ) {}
+
+  ngAfterViewInit() {
+    this.attachBodyContent();
+  }
+
+  private attachBodyContent(): void {
+    this.bodyPortalOutlet?.dispose();
+    const content = this.options.content;
+    if (content instanceof Type) {
+      const componentPortal = new ComponentPortal<T>(
+        content,
+        null,
+        Injector.create({
+          providers: [
+            {
+              provide: DATA,
+              useValue: this.options.contentParams,
+            },
+          ],
+          parent: this.injector,
+        }),
+      );
+      const componentRef =
+        this.bodyPortalOutlet?.attachComponentPortal(componentPortal);
+      Object.assign(componentRef.instance, this.options.contentParams);
+      componentRef.changeDetectorRef.detectChanges();
+    }
+  }
+
+  onAnimation(event: AnimationEvent) {
+    const { phaseName, toState } = event;
+    if (['show', 'hide'].includes(toState)) {
+      const step = [
+        toState,
+        phaseName.charAt(0).toUpperCase() + phaseName.slice(1),
+      ].join('') as Step;
+      this.animationStep$.next(step);
+
+      const backdropElement = this.mask?.nativeElement;
+      if (backdropElement) {
```

Comment:
```suggestion
      if (!backdropElement) {
```

取反判断提前 return，减少嵌套层级